src/NLog/Targets/Wrappers/ConcurrentRequestQueue.cs

```diff
@@ -223,7 +223,7 @@ public override void DequeueBatch(int count, IList<AsyncLogEventInfo> result)
 
             if (dequeueBatch)
             {
-                bool lockTaken = Monitor.TryEnter(_logEventInfoQueue);    // Try to throttle
+                bool lockTaken = result.Count == count && Monitor.TryEnter(_logEventInfoQueue);    // Only try throttle when falling behind
```

Comment:
What will happen if `result.Count` and `count` are not equals, should be below 2 lines 
```cs
for (int i = 0; i < result.Count; ++i)
    Interlocked.Decrement(ref _count);
```
be `Interlocked.Add(ref _count,-count);` instead?

Reply:
Did not know about `Interlocked.Add` with negative delta. Yes that would generate much less overhead, independent of falling behind or not.

Reply:
@snakefoot 
Should it be `-count` instead of `-result.Count` here, if they can be different, to follow the same logic for `!dequeueBatch`

Reply:
I have a mistake, I see it as `i` and think it mean the input list is not empty before add here. So it's ok.

Reply:
I think this a good improvement! Thanks!